os/pm/pm_changestate.c

```diff
@@ -299,6 +299,13 @@ int pm_changestate(int domain_indx, enum pm_state_e newstate)
 		*/
 		pm_changeall(domain_indx, newstate);
 		pdom->state = newstate;
+		
+		/* Adjusting the sleep duration, so that system can revert to NORMAL
+		*  state when next Wifi keep alive signal is required to be sent. 
+		*/
+		if (newstate == PM_SLEEP) {
+			pm_adjust_sleep_duration();
+		}
```

Comment:
I assume that timer types are still as before
i.e. PM_WAKEUP_TIMER and PM_LOCK_TIMER

As per this logic, the changestate function does not differentiate between whether the next wakeup is for the wifi alive signal, or just any other wakeup requested by another APP/source.
It will always adjust the time according to the wifi signal wakeup requirement, but the wakeup may not necessarily be for sending the wifi alive signal only.

You could add a new timer type PM_WIFI_ALIVE_WAKEUP or something and adjust the sleep time in that case only.

Reply:
Hi @amogh-samsung , but I think the operation here is fine, as the wifi keep alive is a signal which have to be sent periodically. Thus, even if other app request for different timer duration, the baseline to wakeup for sending wifi keep alive need to be retained, which I think why we do consideration here so that we won't miss out sending keepalive signal.
Since wifi keep alive signal is a periodic signal, then we should set a periodic timer to handle that~

Reply:
Hi, @edwakuwaku @amogh-samsung  , For now i am just resetting the wake up timer just before sleep every time. So it behaves as periodic .